interface-definitions/service_monitoring_prometheus.xml.in

```diff
@@ -36,6 +36,137 @@
+                              </leafNode>
+                              <leafNode name="ip-protocol-fallback">
+                                <properties>
+                                  <help>Allow fallback to other IP protocol if necessary</help>
+                                  <valueless/>
+                                </properties>
+                              </leafNode>
+                            </children>
+                          </tagNode>
+                        </children>
+                      </node>
+                      <node name="icmp">
+                        <properties>
+                          <help>Configure icmp module</help>
+                        </properties>
+                        <children>
+                          <tagNode name="name">
+                            <properties>
+                              <help>Name of the icmp module</help>
+                            </properties>
+                            <children>
+                              <leafNode name="timeout">
+                                <properties>
+                                  <help>Timeout in seconds for the probe request</help>
+                                  <valueHelp>
+                                    <format>u32:1-60</format>
+                                    <description>Timeout in seconds</description>
+                                  </valueHelp>
+                                  <constraint>
+                                    <validator name="numeric" argument="--range 1-60"/>
+                                  </constraint>
+                                  <constraintErrorMessage>Timeout must be between 1 and 60 seconds</constraintErrorMessage>
+                                </properties>
+                                <defaultValue>5</defaultValue>
+                              </leafNode>
+                              <leafNode name="preferred-ip-protocol">
+                                <properties>
+                                  <help>Preferred IP protocol for this module</help>
+                                  <valueHelp>
+                                    <format>ip4</format>
+                                    <description>Prefer IPv4</description>
+                                  </valueHelp>
+                                  <valueHelp>
+                                    <format>ip6</format>
+                                    <description>Prefer IPv6</description>
+                                  </valueHelp>
+                                  <constraint>
+                                    <regex>(ip4|ip6)</regex>
+                                  </constraint>
+                                </properties>
+                                <defaultValue>ip6</defaultValue>
+                              </leafNode>
+                              <leafNode name="ip-protocol-fallback">
+                                <properties>
```

Comment:
Looks like a XML definitions that's used two times for blackbox exporter.
That would suffice for an includable and reusable XML building block.